dali/pipeline/executor/executor_impl.cc

```diff
@@ -664,10 +664,11 @@ void Executor<WorkspacePolicy, QueuePolicy>::InitIterationData() {
   size_t iteration_data_size = CalcIterationDataSize();
   iteration_data_.resize(iteration_data_size);
   for (auto& id : iteration_data_) {
-    id.iter_data = std::make_shared<IterationData>();
+    id = std::make_shared<IterationData>();
     if (checkpointing_) {
+      id->checkpoint = std::make_shared<Checkpoint>();
```

Comment:
Is it a problem that `checkpoint` would be NULL when `checkpointing_ == false`?

Reply:
I don't think so, @stiepan ?

Reply:
Me neither.